client/src/org/compiere/grid/ed/VPAttribute.java

```diff
@@ -475,7 +475,7 @@ public void actionPerformed(ActionEvent e)
 			if (M_AttributeSet_ID != 0)
 			{
 				MAttributeSet mas = MAttributeSet.get(Env.getCtx(), M_AttributeSet_ID);
-				exclude = mas.excludeEntry(m_AD_Column_ID, Env.isSOTrx(Env.getCtx(), m_WindowNo));
+				exclude = mas.excludeEntry(product.get_Table_ID(), Env.isSOTrx(Env.getCtx(), m_WindowNo));

```

Comment:
Suggest MColumn.getTable_ID(Env.getCtx(), m_AD_Column_ID, null) instead.  Product.get_Table_ID() will return the product table rather than the table associated with the column.

Reply:
Hi, maybe is better use I_M_Product.Table_ID constant

Reply:
Hi, maybe is better use I_M_Product.Table_ID constant

Reply:
I reviewed and implement other approaches, so that method singer change and now receipt only the table id

On the other hand, set the same logic that ZK UI to(boolean productWindow ).

Test executed :

1.- Product Window TShirt click in Attribute Set Instance field and show the Attribute Dialog
2.-Adding to Attribute Set Fertilizer Lot exclude the M_InOutLine table with IsSOTrx = 'N'
3.-Material Receipt Window add a line for Fertilizer#50 and click  Attribute Set Instance field and the  Attribute Dialog is not  open, this is ok
4.-Shipment Window add a line for Fertilizer#50 and click  Attribute Set Instance field and the  Attribute Dialog is open, this is ok

Reply:
Still think the table is not the product table.  It should be the table of the tab using the VPAttribute dialog.  It is is used to check excludes by table, so it would need to be the C_OrderLine table on an order or the M_InOutLine table on a shipment receipt.

Reply:
Hi @mckayERP, you can see that this code only check if the column is the product window 

https://github.com/adempiere/adempiere/pull/3221/files#diff-31087b85b9e8a767119764d848dc08f7R474
Here you can see that now use the column id to solve of table

https://github.com/adempiere/adempiere/pull/3221/files#diff-31087b85b9e8a767119764d848dc08f7R485